src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -484,6 +489,307 @@ public static string ConvertToJson(object objectToProcess, in ConvertToJsonConte
             }
         }
 
+        // The implementation is the same as JavaScriptEncoder.Default but can be customized.
+        // Default JavaScriptEncoder always escape HTML and follow codepoints (the comment come from .Net Core):
+        // 1. Forbid codepoints which aren't mapped to characters or which are otherwise always disallowed
+        //    (includes categories Cc, Cs, Co, Cn, Zs [except U+0020 SPACE], Zl, Zp)
+        // 2. '\' (U+005C REVERSE SOLIDUS) must always be escaped in Javascript / ECMAScript / JSON.
+        //    '/' (U+002F SOLIDUS) is not Javascript / ECMAScript / JSON-sensitive so doesn't need to be escaped.
+        // 3. '`' (U+0060 GRAVE ACCENT) is ECMAScript-sensitive (see ECMA-262).
+        private static JavaScriptEncoder s_escapeNonAsciiEncoder = InitEscapeNonAsciiEncoder();
+        private static JavaScriptEncoder InitEscapeNonAsciiEncoder()
+        {
+            var textEncoderSettings = new TextEncoderSettings(UnicodeRanges.BasicLatin);
+
+            return JavaScriptEncoder.Create(textEncoderSettings);
+        }
+
+        /// <summary>
+        /// Convert an object to JSON string.
+        /// </summary>
+        public static string ConvertToJson2(object objectToProcess, in ConvertToJsonContext context)
+        {
+            if (objectToProcess is null)
+            {
+                return "null";
+            }
+
+            try
+            {
+                var options = new JsonSerializerOptions()
+                {
+                    WriteIndented = !context.CompressOutput,
+                    MaxDepth = context.MaxDepth,
+                    IgnoreNullValues = false,
+                    Encoder = context.StringEscapeHandling switch
+                    {
+                        StringEscapeHandling.Default => JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+                        StringEscapeHandling.EscapeHtml => JavaScriptEncoder.Default,
+                        StringEscapeHandling.EscapeNonAscii => s_escapeNonAsciiEncoder,
+                        _ => JavaScriptEncoder.Default
+                    }
+                };
+
+                if (context.EnumsAsStrings)
+                {
+                    options.Converters.Add(new System.Text.Json.Serialization.JsonStringEnumConverter());
+                }
+
+                options.Converters.Add(new JsonStringEnumConverter64());
```

Comment:
Add JsonStringEnumConverter64 if EnumAsStrings is True?

Reply:
EnumAsStrings is in line 535. JsonStringEnumConverter64 is old PowerShell workaround (see comment there).